src/main/java/frc/robot/util/LogManager.java

```diff
@@ -1,38 +1,111 @@
  * View logs using MechanicalAdvantage's advantage scope (<a href="https://github.com/Mechanical-Advantage/AdvantageScope">...</a>)
  */
-public class LogManager {
-
-  public static DataLog DATA_LOG = DataLogManager.getLog();
+public class LogManager extends DogLog {
 
   private static final ArrayList<Log<?>> logs = new ArrayList<>();
 
-  public static <T> void add(Log<T> log) {
+  /**
+   * Log a Log object periodically
+   * @param <T> Type of item being logged
+   * @param log The Log object to log
+   */
+  public static <T> void log(Log<T> log) {
     logs.add(log);
   }
+  
+  /**
+   * Log a supplier every 20ms
+   * @param <T> Type of item being logged
+   * @param name Name (key) of item being logged
+   * @param value Supplier for value being logged 
+   */
+  public static <T> void logSupplier(String name, Supplier<T> value) {
+    log(new Log<>(name, value));
+  }
 
-  public static <T> void add(String name, T value) {
-    add(name, ()->value);
+    /**
+   * Log a supplier every 20ms with a minimum and a maxium, and if the value is outside of the min or max, it will log a fault.
+   * @param <T> Type of item being logged
+   * @param name Name (key) of item being logged
+   * @param value Supplier for value being logged 
+   * @param min The minimum that the value can be
+   * @param min The maximum that the value can be
+   */
+  public static <T> void logSupplier(String name, Supplier<T> value, T min, T max) {
+    log(new Log<>(name, value, min, max));
   }
-  public static <T> void add(String name, Supplier<T> value) {
-    add(new Log<>(name, value));
+
+
+  /**
+   * Log a supplier periodically
+   * @param <T> Type of item being logged
+   * @param name Name (key) of item being logged
+   * @param value Supplier for value being logged 
+   * @param updateDelay The amount of time, in milliseconds, between logs
+   */
+  public static <T> void logSupplier(String name, Supplier<T> value, int updateDelay) {
+    log(new Log<>(name, value, updateDelay));
   }
 
-  public static <T> void add(String name, Supplier<T> value, Duration duration) {
-    add(new Log<>(name, value, duration));
+    /**
+   * Log a supplier periodically with a minimum and a maxium, and if the value is outside of the min or max, it will log a fault. 
+   * @param <T> Type of item being logged
+   * @param name Name (key) of item being logged
+   * @param value Supplier for value being logged 
+   * @param updateDelay The amount of time, in milliseconds, between logs
+   * @param min The minimum that the value can be
+   * @param min The maximum that the value can be
+
+   */
+  public static <T> void logSupplier(String name, Supplier<T> value, int updateDelay, T min, T max) {
+    log(new Log<>(name, value, updateDelay, min, max));
   }
 
+
+  public static void log(String name, int value, int min, int max) {
+    log(name, value);
+    
+    if (value <= min || value >= max) {
+      logFault(name + " is out of specified range!");
+    }
```

Comment:
It should be < and >, not <= and >=. Also, maybe log the value that was out of range in addition tot he name.